src/Commands/Integration.ts

```diff
@@ -76,30 +76,26 @@ export class Integration implements CommandModule {
       })
       .middleware((args: Arguments) => {
         container
-          .register(IntegrationOptions, {
+          .register<IntegrationOptions>(IntegrationOptions, {
             useValue: {
-              insecure: args.insecure,
-              timeout: Number(args.timeout),
-              baseUrl: args.baseUrl,
-              user: args.user,
-              apiKey: args.password
+              // TODO: (victor.polyakov@neuralegion.con) Write correct type checking
+              timeout: +(args.timeout as number),
+              apiKey: args.password as string,
+              user: args.user as string,
+              baseUrl: args.baseUrl as string,
+              insecure: args.insecure as boolean
             }
           })
-          .register(RabbitMQBusOptions, {
+          .register<RabbitMQBusOptions>(RabbitMQBusOptions, {
             useValue: {
               exchange: 'EventBus',
               clientQueue: `integration:${args.accessKey}`,
               connectTimeout: 10000,
               url: args.bus as string,
-              proxyUrl: args.proxy as string,
+              proxyUrl: (args.proxyExternal ?? args.proxy) as string,
               credentials: {
                 username: 'bot',
                 password: args.token as string
-              },
-              onError(e: Error) {
-                clearInterval(timer);
-                logger.error(`Error during "integration": ${e.message}`);
-                process.exit(1);
```

Comment:
It is not clear to me. Why did you remove it?

Reply:
It is very interesting, though. If you turn on strictNullCheck, you can se that the type RabbitMQBusOptions doesn't have additional param with type like onError. There aren't any other place where use data with that callback. So I think, it's a mistake.